content/curating/guidelines/kata.md

```diff
@@ -0,0 +1,45 @@
+---
+sidebar: curating
+prev: /curating/guidelines/
+next: /curating/guidelines/translation/
+---
+
+
+# Kata Curating Guidelines
+
+This article contains a set of guidelines, which can be used by reviewers to determine whether a kata meets quality requirements to be approved and become available for users to train on. They were collected to help ensure that kata are of sufficient quality and users' experience will be as good as possible.
+
+## Review
+
+- **The kata must be reviewed before it's approved.** See [Curating a Kata HOWTO][howto-review-kata] for more details on how to review a kata.
+- **The review should be as complete as possible,** including all applicable steps mentioned in [Curating a Kata HOWTO][howto-review-kata]. If the reviewer is not able to perform a complete review, they should ask other users for help.
+- **Conformity to [kata authoring guidelines][guidelines-authoring-kata] should be verified.** All applicable guidelines should be followed by the author.
+
+
+## Feedback
+
+- **The satisfaction rating should indicate quality of kata as a concept, not its current state of condition in terms of open issues which can be fixed.** If you think the kata itself is valuable but you simply have some issues with it, then you should select "Somewhat" and create an issue instead. If you think the kata is really great, other than the issues, then you can choose "Very". Do not choose "None" simply because the kata has an issue but it is otherwise a good kata.
+- **When casting a vote for difficulty rating, reviewers should consider a broader picture than just their own skills.** It's true that the way how difficulty rankings are handled by Codewars is far from perfect, and not balanced well. It's also true that difficulty level is not easy to estimate, because what is easy for one, can be difficult for another, and vice versa. The way how difficulty ranking works will be hopefully improved in the future, but until then, your votes should not throw the system deliberately off-balance.
+- Posts which provide information on issues encountered during the review **should contain information helpful for fixing them.** Feedback of the form _"This and that is wrong."_ is not very constructive nor helpful to the author. Feedback post should provide some guidance, hints, or links to articles explaining how to fix the problems.
```

Comment:
Similar to this line, but we should encourage efficient communication.

When communicating asynchronously in general, we should minimize the number of required exchanges. Otherwise, a lot of time will be wasted on both ends.

I've seen too many of this:
```text
A: the user can modify the input
(5 minutes later)
B: hmm, why is that a problem?
(3 minutes later)
A: (sigh) because ...
(30 minutes later)
B: understood. I tried, but couldn't figure out. Can you show me how to fix it?
(5 minutes later)
A: (thinks B is incompetent) you're not good enough. just do this and that!
```

To minimize the number of required exchanges, you need to **think about the person you're writing to** and **optimize the message for them**. The message should contain just enough information to get your point across.

For the above example, it's obvious (even more obvious when it's their first kata) that the author doesn't know about the effect of the user mutating the input. Even if they did know and simply had forgotten it, including relevant information doesn't hurt. It's also likely that the author is not familiar with the idea and might need your help.

If you considered these, you should end up writing the following to minimize the required exchange/effort:
```text
A: Hey, the current setup allows users to modify the input. Because ... see ...
(5 minutes later)
B: Thanks for the info. I'll look into it and let you know when it's fixed.
(10 minutes later)
B: Hey, I think it's fixed now, can you confirm?
(5 minutes later)
A: Yup, it's fixed :)
```
Some might think this will be more work for them, but it's usually much less work considering the additional exchanges that was required in the first example.

1. Try to include all the necessary information required for them to understand you.
2. Remove unnecessary remarks, especially those that can be inflammatory and cause mental/emotional overhead for the reader.
3. Optimize for them and not yourself.

Also, be always respectful and nice to others. If you need a reason to be nice, it's because it's the most efficient.

Reply:
I think you have a typo, it should be `C: (sigh) because ...`. It's never the `A`.

But jokes aside, I know what you mean and I was wondering how to tackle this issue. What I plan is to create some more or less language-dedicated pages with tutorial/description how to create kata in a way so they stick to the guidelines and avoid most common pitfalls. I get it why As go they way they do, and as much as I don't like it, I think they need some support too, so ideally, their review could look like: 

`A: [user can modify input](/languages/python/creating-kata/#input-mutation)`

I wanted to create such thing for Python or JS as the most popular languages ATM, and see how it turns out.

Reply:
I want to keep guidelines themselves short, and I think the way how this point is now, is OK (unless you have some language related remarks). I've added additional paragraphs on review comments in "Curating howto" parts [for kata](https://deploy-preview-185--reverent-edison-2864ea.netlify.app/curating/kata/#review-comments) and for [translations](https://deploy-preview-185--reverent-edison-2864ea.netlify.app/curating/translation/#review-comments). They can be expanded there, but I am not sure if it's needed. Let me know.

I do not like the explicit "be nice" or "respect others", because I personally think they are useless against offenders, and annoying for those who follow them anyway. "Requirement cannot be enforced by virtue", right? ;) But I can add something like this if you really want it.

Reply:
> `A: [user can modify input](/languages/python/creating-kata/#input-mutation)`

Yeah, I was thinking providing a way to insert those for common issues for beta kata. Or even provide dedicated buttons for them. Or something similar to issue templates on GitHub.

> I do not like the explicit "be nice" or "respect others", because I personally think they are useless against offenders, and annoying for those who follow them anyway.

I agree having "be nice" here won't help. But there are communication styles you should consider when reviewing someone else's work to minimize unnecessary overhead and I think that's worth mentioning somewhere. Like prefer "**the test** should" over "**you** should". Using "**you**" tend to make the author defensive and take the comment personally. There are many articles about this topic.

It's more of a tip than a requirement and also related to #142. We can include it there and link to it.

Reply:
Oh I think I finally get what you mean, and somehow it never occurred to me that Codewars content reviews can be held in a tone similar to actual code review. Yay this makes the thing even cooler. I will think how to put it in the docs.

I would appreciate any feedback on general shape of the docs related to curating and authoring. Please share any remarks you'd have, either here or in related discussion thread (#190). Thanks a lot.

Reply:
Exactly. I believe this is a **critical** skill to have if you're working with people (almost any profession and activities within communities). No one wants to _continuously_ work with you if you don't know how to adjust your communication style.

This is why some long-time users end up getting blocked by some authors. Even if they meant well, they failed to communicate their intention.

I think we should expand on this later with more specific examples for common issues, but this is enough for now.